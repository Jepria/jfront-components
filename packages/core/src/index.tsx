export * from '@jfront-ui/app'
export * from '@jfront-ui/datepicker'
export * from '@jfront-ui/form'
//export * from '@jfront-ui/grid'
export * from '@jfront-ui/hooks'
//export * from '@jfront-ui/input'
export * from '@jfront-ui/layout'
//export * from '@jfront-ui/pagingbar'
//export * from '@jfront-ui/tabpanel'
//export * from '@jfront-ui/toolbar'
