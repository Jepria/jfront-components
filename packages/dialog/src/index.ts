export * from './SimpleDialog'
export * from './AlertDialog'
export * from './FormDialog'
export * from './ErrorDialog'
export * from './WarningDialog'