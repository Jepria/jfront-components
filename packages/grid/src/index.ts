export * from './common'
export * from './jepgrid'
