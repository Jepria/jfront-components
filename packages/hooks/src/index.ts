export * from "./useOnClickOutside"
export * from "./useRouteMapping"
export * from "./useTree/"
export * from "./useTree/types"
export * from "./useFilter"
