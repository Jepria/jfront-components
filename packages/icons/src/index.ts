export * from "./Add"
export * from "./Delete"
export * from "./Edit"
export * from "./Exclamation"
export * from "./First"
export * from "./Last"
export * from "./Loading"
export * from "./Next"
export * from "./OpenIcon"
export * from "./Prev"
export * from "./Refresh"
export * from "./Save"
export * from "./Search"
export * from "./SettingsSmall"
export * from "./Split"
export * from "./View"
