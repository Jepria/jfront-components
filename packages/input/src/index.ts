export interface InputProps {
  error?: string
  name?: string
  isLoading?: boolean
}

export * from "./SelectInput"
export * from "./TextInput"
export * from "./NumberInput"
export * from "./MaskedInput"
export * from "./MoneyInput"
export * from "./DecimalInput"
export * from "./MoneyInput"
