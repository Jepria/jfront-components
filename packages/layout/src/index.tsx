import styled from 'styled-components';

export const FormContainer = styled.div`
  padding: 5px;
`;

export const VerticalLayout = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
`;

export const HorizontalLayout = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
`;

export const Page = styled.div`
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
`;

export const Header = styled.header`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`;

export const Content = styled.section`
  box-sizing: border-box;
  width: 100%;
  overflow: auto;
  -webkit-box-flex: 1;
      -ms-flex: 1;
          flex: 1;
  -ms-flex-positive: 1;
      flex-grow: 1;
`;

export const Footer = styled.footer`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`;
