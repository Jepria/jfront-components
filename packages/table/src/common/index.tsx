import React, {
  HTMLAttributes,
  TableHTMLAttributes,
  RefAttributes,
} from "react"
import styled from "styled-components"
import { TablePagingBar, TablePagingBarProps } from "./TablePagingBar"
import { TableBody } from "./TableBody"
import { Table as StyledTable } from "./Table"
import {
  TableHeader,
  TableHeaderCell,
  TableHeaderCellProps,
} from "./TableHeader"
import { TableRow, TableCell, TableRowProps, TableCellProps } from "./TableRow"

export * from "./Table"

export type Table = React.FC<TableHTMLAttributes<HTMLTableElement>> & {
  Table: React.FC<
    TableHTMLAttributes<HTMLTableElement> & RefAttributes<HTMLTableElement>
  >
  Header: React.FC<HTMLAttributes<HTMLTableSectionElement>>
  HeaderCell: React.FC<TableHeaderCellProps>
  Body: React.FC<
    React.DetailedHTMLProps<
      React.HTMLAttributes<HTMLTableSectionElement>,
      HTMLTableSectionElement
    >
  >
  Row: React.FC<TableRowProps>
  Cell: React.FC<TableCellProps>
  PagingBar: React.FC<TablePagingBarProps>
}

const Container = styled.section`
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
`

export const Table: Table = (props) => {
  return <Container {...props}>{props.children}</Container>
}

Table.Table = (props) => {
  return <StyledTable {...props}>{props.children}</StyledTable>
}

Table.Header = (props) => {
  return <TableHeader {...props}>{props.children}</TableHeader>
}

Table.HeaderCell = (props) => {
  return <TableHeaderCell {...props}>{props.children}</TableHeaderCell>
}

Table.Body = (props) => {
  return <TableBody {...props}>{props.children}</TableBody>
}

Table.Row = (props) => {
  return <TableRow {...props}>{props.children}</TableRow>
}

Table.Cell = (props) => {
  return <TableCell {...props}>{props.children}</TableCell>
}

Table.PagingBar = (props: TablePagingBarProps) => {
  return <TablePagingBar {...props} />
}
