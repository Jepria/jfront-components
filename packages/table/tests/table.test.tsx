import { render } from "@testing-library/react"
import React from "react"
import { Table } from "../src"

test("Checking for the existence of an element Table", () => {
  render(<Table />)
})
