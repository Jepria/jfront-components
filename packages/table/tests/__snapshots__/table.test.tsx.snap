// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot  1`] = `
<DocumentFragment>
  <section
    class="sc-fzoLag inWPiQ"
  >
    <div
      class="sc-fznZeY JotLz"
    >
      <table
        class="sc-fznKkj fMNvPW"
      >
        <thead
          class="sc-fzokOt dNeaSo"
        >
          <tr
            class="sc-fzqNJr jQXiND"
          >
            <th
              class="sc-fzqBZW cVRUvW"
            >
              header1
            </th>
            <th
              class="sc-fzqBZW cVRUvW"
            >
              header2
            </th>
            <th
              class="sc-fzqBZW cVRUvW"
            >
              header3
            </th>
            <th
              class="sc-fzqBZW cVRUvW"
            >
              header4
            </th>
          </tr>
        </thead>
        <tbody
          class="sc-fzpans STsDJ"
        >
          <tr
            class="sc-fzqNJr jQXiND"
          >
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row1
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row1
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row1
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row1
            </td>
          </tr>
          <tr
            class="sc-fzqNJr jQXiND"
          >
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row2
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row2
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row2
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row2
            </td>
          </tr>
          <tr
            class="sc-fzqNJr jQXiND"
          >
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row3
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row3
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row3
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row3
            </td>
          </tr>
          <tr
            class="sc-fzqNJr jQXiND"
          >
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row4
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row4
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row4
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row4
            </td>
          </tr>
          <tr
            class="sc-fzqNJr jQXiND"
          >
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row5
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row5
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row5
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row5
            </td>
          </tr>
          <tr
            class="sc-fzqNJr jQXiND"
          >
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row6
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row6
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row6
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row6
            </td>
          </tr>
          <tr
            class="sc-fzqNJr jQXiND"
          >
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row1
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row1
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row1
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row1
            </td>
          </tr>
          <tr
            class="sc-fzqNJr jQXiND"
          >
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row2
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row2
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row2
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row2
            </td>
          </tr>
          <tr
            class="sc-fzqNJr jQXiND"
          >
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row3
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row3
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row3
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row3
            </td>
          </tr>
          <tr
            class="sc-fzqNJr jQXiND"
          >
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row4
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row4
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row4
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row4
            </td>
          </tr>
          <tr
            class="sc-fzqNJr jQXiND"
          >
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row5
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row5
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row5
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row5
            </td>
          </tr>
          <tr
            class="sc-fzqNJr jQXiND"
          >
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row6
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row6
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row6
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row6
            </td>
          </tr>
          <tr
            class="sc-fzqNJr jQXiND"
          >
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row1
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row1
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row1
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row1
            </td>
          </tr>
          <tr
            class="sc-fzqNJr jQXiND"
          >
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row2
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row2
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row2
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row2
            </td>
          </tr>
          <tr
            class="sc-fzqNJr jQXiND"
          >
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row3
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row3
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row3
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row3
            </td>
          </tr>
          <tr
            class="sc-fzqNJr jQXiND"
          >
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row4
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row4
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row4
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row4
            </td>
          </tr>
          <tr
            class="sc-fzqNJr jQXiND"
          >
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row5
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row5
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row5
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row5
            </td>
          </tr>
          <tr
            class="sc-fzqNJr jQXiND"
          >
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row6
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row6
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row6
            </td>
            <td
              class="sc-fzoyAV ffnEzq"
            >
              Row6
            </td>
          </tr>
          <tr
            class="sc-fzplWN hbEGsR"
          >
            <td
              class="sc-fznyAO eMIraO"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </section>
</DocumentFragment>
`;
